### ISSUE TITLE
mlvpn client not working on OpenBSD

### ISSUE THREAD

**Issue by rapenne-s** [zehome/MLVPN#127]
Hi,

I've built last mlvpn version on OpenBSD, running the server seems to work as it waits for packets, but when I run the client, I get some output then mlvpn stops.

```
solene@laptop ~ $ doas mlvpn -c /etc/mlvpn/mlvpn.conf  -u _mlvpn -v --debug 
2019-12-02T07:50:06 [ DBG] absolute maximum mtu: 1444
2019-12-02T07:50:06 [INFO/config] new password set
2019-12-02T07:50:06 [INFO/config] dsl1 tunnel added
2019-12-02T07:50:07 [INFO] created interface `tun0'
2019-12-02T07:50:07 [INFO] dsl1 bind to *
solene@laptop ~ $
```

I use this configuration

```
[general]
statuscommand = "/etc/mlvpn/mlvpn_updown.sh"
mode = "client"
mtu = 1444
tuntap = "tun"
interface_name = "tun0"
timeout = 30
password = "XXX"
cleartext_data = 0

[dsl1]
bindhost = "*"
remotehost = "6.5.4.2"
remoteport = 5080
```
Am I doing things wrong or is there an hidden bug, if so how can I help debugging it?

**Comment by zehome** [comment]
Yes, I came across this issue too. Will debug it tonight

**Comment by zehome** [comment]
http://openbsd-archive.7691.n7.nabble.com/net-mlvpn-ioctl-TUNSIFMODE-Invalid-argument-td378478.html

**Comment by zehome** [comment]
I've fixed it in b6182155577fb661529f386731cc276f0c8bed1c I'll post a patch for the OpenBSD ports

**Comment by zehome** [comment]
This is fixed in OpenBSD cvs https://cvsweb.openbsd.org/ports/net/mlvpn/
